### PR TITLE
Updting to FH-3.1.0-SNAPSHOT SDK

### PR DESCRIPTION
**Pull request by matzew**
Done for FH-2647

**DO NOT MERGE**
@secondsun @matskiv Here is an _early_ (but mostly complete) update of the template app.
I have ONLY built it locally (using local repo, see gradle file)

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -33,8 +33,12 @@ license {
     strictCheck true
 }
 
+repositories {
+    mavenLocal()
+}
```
Comment (matzew) [inline app/build.gradle:38]: yes.... :-) that's me being lazy

File: app/build.gradle [hunk 2]
```diff
@@ -33,8 +33,12 @@ license {
...
+
 dependencies {
-    compile 'com.feedhenry:fh-android-sdk:3.0.0'
-    compile 'com.android.support:appcompat-v7:23.1.1'
-    compile 'org.jboss.aerogear:aerogear-android-push:2.2.2'
+    compile 'com.feedhenry:fh-android-sdk:3.1.0-SNAPSHOT'
```
Comment (matzew) [inline app/build.gradle:41]: will be a non-snapshot dependency...

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -61,20 +61,32 @@
...
             android:permission="com.google.android.c2dm.permission.SEND">
             <intent-filter>
                 <action android:name="com.google.android.c2dm.intent.RECEIVE" />
-                <action android:name="com.google.android.c2dm.intent.REGISTRATION" />
-
-                <category android:name="org.feedhenry.pushstarter" />
             </intent-filter>
             <meta-data
                 android:name="DEFAULT_MESSAGE_HANDLER_KEY"
                 android:value="org.jboss.aerogear.unifiedpush.helloworld.handler.NotificationBarMessageHandler" />
```
Comment (matzew) [inline app/src/main/AndroidManifest.xml]: @secondsun why does this have the community package? I'd guess this is wrong, but app works 😁

**Comment by matzew** [comment]
On https://github.com/fheng/fhcap/pull/1963 I already mentioned that I have tested an _existing_ template app with the 'updated' server.

Before testing the app update of this PR, I had one device, with legacy token:
 

After updating the app on the device, it registers fine, I have two tokens, the newly (InstanceID) formatted token, which contains a `:` and a legacy token, both for the same device. As expected:
 

Sending a push, I see that exactly one message came in 😄 
 

On the server side I did extract this log, which shows it is performing a send to
- global topic
- one (legacy) device token

All as expected. See the log for details:

<pre>
08:05:51,718 INFO  [org.jboss.aerogear.unifiedpush.message.NotificationRouter] (http-/0.0.0.0:9081-4) Internal dispatching of push message for one android variant
08:05:51,806 INFO  [org.jboss.aerogear.unifiedpush.message.token.TokenLoader] (Thread-1014 (HornetQ-client-global-threads-200466920)) Preparing message delivery and loading tokens for the ANDROID 3rd-party Push Network (for 1 variants)
08:05:51,838 INFO  [org.jboss.aerogear.unifiedpush.message.token.TokenLoader] (Thread-1014 (HornetQ-client-global-threads-200466920)) Loaded batch #1, containing 1 tokens, for android variant (6e9faf7c-35b0-4f8f-b0ef-586b9b88e086)
08:05:51,840 INFO  [org.jboss.aerogear.unifiedpush.message.NotificationDispatcher] (Thread-1017 (HornetQ-client-global-threads-200466920)) Received UnifiedPushMessage from JMS queue, will now trigger the Push Notification delivery for the android variant (6e9faf7c-35b0-4f8f-b0ef-586b9b88e086)
<b>08:05:51,845 INFO  [org.jboss.aerogear.unifiedpush.message.sender.GCMPushNotificationSender] (Thread-1017 (HornetQ-client-global-threads-200466920)) Sent push notification to GCM topic: &#x2F;topics&#x2F;6e9faf7c-35b0-4f8f-b0ef-586b9b88e086</b>
08:05:51,900 INFO  [org.jboss.aerogear.unifiedpush.message.token.TokenLoader] (Thread-1014 (HornetQ-client-global-threads-200466920)) Loaded batch #2, containing 1 tokens, for android variant (6e9faf7c-35b0-4f8f-b0ef-586b9b88e086)
08:05:51,913 INFO  [org.jboss.aerogear.unifiedpush.message.NotificationDispatcher] (Thread-1015 (HornetQ-client-global-threads-200466920)) Received UnifiedPushMessage from JMS queue, will now trigger the Push Notification delivery for the android variant (6e9faf7c-35b0-4f8f-b0ef-586b9b88e086)
<b>08:05:51,916 INFO  [org.jboss.aerogear.unifiedpush.message.sender.GCMPushNotificationSender] (Thread-1015 (HornetQ-client-global-threads-200466920)) Sent push notification to GCM Server for 1 registrationIDs</b>
08:05:52,254 INFO  [org.jboss.aerogear.unifiedpush.message.sender.GCMPushNotificationSender] (Thread-1015 (HornetQ-client-global-threads-200466920)) Processing [NotRegistered] error code from GCM response, for registration ID: [APA91bH8zCY1UBji5rBP3jLUmuUNVkhzh9mW2a0GofclKjC6TO14_ao7M69xBQiL0SDJ_AfqKdajpf5WHMsWl9zmhUi0RPkMYWr0T8SjDO9hinlITZFkMMKIJvinN85wYpKk0VKli4TB]
<b>08:05:52,257 INFO  [org.jboss.aerogear.unifiedpush.message.sender.GCMPushNotificationSender] (Thread-1015 (HornetQ-client-global-threads-200466920)) Based on GCM response data and error codes, deleting 1 invalid or duplicated Android installations</b>
</pre>

The log also shows, that the legacy token has been _DELETED_ because it's an invalid (or duplicated) token for the device, based on the old version of the app, that used to _NOT_ use InstanceID for GCM tokens. 

@secondsun @matskiv  let me know what you think...

Again, please do not merge, as the FH-android-sdk is not yet published/released 😬

File: app/src/main/AndroidManifest.xml (hunk 3 above)
Comment (secondsun) [inline app/src/main/AndroidManifest.xml]: Was an artifact from when the application was ported probably.

**Comment by secondsun** [comment]
+1 on this PR assuming 3.1.0 contains agdroid-push-3.0.1 instead of 3.0.0 (see https://github.com/aerogear/aerogear-android-push/pull/61)

**Comment by matskiv** [comment]
:+1: Works fine and builds on BF without any problem.

File: app/src/main/AndroidManifest.xml (hunk 3 above)
Comment (matzew) [inline app/src/main/AndroidManifest.xml]: @secondsun can we change to to... some random value? e.g.

```
org.feedhenry.pushstarter.handler.NotificationBarMessageHandler
```

Or does it have to match something ?

Comment (secondsun) [inline app/src/main/AndroidManifest.xml]: It just has to match the class name.  We can rename it to whatever.

**Comment by secondsun** [comment]
LGTM

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -85,7 +85,7 @@
             </intent-filter>
             <meta-data
                 android:name="DEFAULT_MESSAGE_HANDLER_KEY"
-                android:value="org.jboss.aerogear.unifiedpush.helloworld.handler.NotificationBarMessageHandler" />
+                android:value="org.feedhenry.pushstarter.handler.NotificationBarMessageHandler" />
         </service>
```
Comment (matzew) [inline app/src/main/AndroidManifest.xml:89]: @secondsun I actually changed two packages here :-)

**Pull request closed without merging**